### DISCUSSION TITLE
Why does the Asset.Builder overwrites properties?

### DISCUSSION THREAD

**Discussion by DominikPinsel** [eclipse-edc/Connector#750] [Q&A]
The `properties` method of the `Asset.Builder` overwrites all previous set properties.

For example this asset is created without an id
```java
Asset asset = Asset.Builder.newInstance()
                .id(id)
                .properties(properties)
                .build();
```

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

**Comment by paullatzelsperger** [comment]
that's expected behaviour. if you set all the properties, they get overwritten.

**Comment by ndr-brt** [comment]
I agree that the api of this builder can bring to unexpected results. 
Should the `properties` method just add all the properties instead of overwrite the whole map?

**Comment by DominikPinsel** [comment]
Yea. I would assume that when I set the id in the builder, that the id in the builder is set. The only thing in a builder that should be overwrite this would be setting the id another time.
